conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -195,22 +202,22 @@ def _get_required_components_cpp(self):
                                                 self.is_host_windows)
 
             public_comp_deps = []
-            for require in comp.requires:
-                if "::" in require:  # Points to a component of a different package
-                    pkg, cmp_name = require.split("::")
```

Comment:
Trying to remove a bit the "parsing" of ``cpp_info.requires`` from the Generators. I have tried to go further in the abstraction, but failed.